File: Production Back/Production.Domen/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Production.Domen
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string ProfessionalName { get; set; }
        public string Shape { get; set; }
        public string Description { get; set; }
        public string Instruction { get; set; }
        public double Price { get; set; }
        public int UnitOfMeasureID { get; set; }
        public UnitOfMeasure UnitOfMeasure { get; set; }

        public static implicit operator List<object>(Product v)
        {
            throw new NotImplementedException();
        }
    }
}

File: Production-SPA/src/app/models/product.ts
import { UnitOfMeasure } from '../models/unit-of-measure';

export class Product {
  id: number;
  name: string;
  professionalName: string;
  shape: string;
  description: string;
  instruction: string;
  price: number;
  unitOfMeasureID: number;
  unitOfMeasure: UnitOfMeasure;

  constructor(p?: any) {
    if (p == null) {
      return;
    }
    this.id = p.id;
    this.name = p.name;
    this.professionalName = p.professionalName;
    this.shape = p.shape;
    this.description = p.description;
    this.instruction = p.instruction;
    this.price = p.price;
    this.unitOfMeasureID = p.unitOfMeasureID;
    this.unitOfMeasure = new UnitOfMeasure(p.UnitOfMeasure);
  }
}
